### PR TITLE
fix: users migration from 7.41.2 to 7.49.0 on OpenShift

### PR DESCRIPTION
**Pull request by tolusha**
Signed-off-by: Anatolii Bazko <abazko@redhat.com>

## What does this pull request change?
* Fetch ALL keycloak users by pages
* Check if user exists in `dbche` database before inserting its profile
* Escaping `'` in user first and last name

## What issues does this pull request fix or reference?
https://issues.redhat.com/browse/CRW-3282

## How to test this PR
1. Deploy CRW on OpenShift
2. Create around 200 users in keycloak and OpenShift
3. Login for 10 users (to add them to dbche)
4. check `codeready-users.txt` file, it must be at least 200 records
5. run migration, check that only 10 users migrated

## Specify the version of the product this pull request applies to
7.49.x 7.50.x 7.51.x 7.52.x

## Pull Request checklist

The author and the reviewers validate the content of this pull request with the following checklist, in addition to the [automated tests](code_review_checklist.adoc).

- Any procedure:
  - [ ] Successfully tested.
- Any page or link rename:
  - [ ] The page contains a redirection for the previous URL.
  - Propagate the URL change in:
    - [ ] Dashboard [default branding data](https://github.com/eclipse-che/che-dashboard/blob/main/packages/dashboard-frontend/src/services/bootstrap/branding.constant.ts)
    - [ ] Chectl [constants.ts](https://github.com/che-incubator/chectl/blob/main/src/constants.ts)
- [ ] Builds on [Eclipse Che hosted by Red Hat](https://workspaces.openshift.com).
- [ ] the *`Validate language on files added or modified`* step reports no vale warnings.

### CODE REVIEW

**Comment by dmytro-ndp** [comment]
@tolusha : rollback.sh script has failed being run just after `1-prepare.sh`, and CRW application remained not available:
```
[INFO] New subscription starting CSV: crwoperator.v2.15.4-0.1658786714.p.
Warning: resource subscriptions/codeready-workspaces is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by oc apply. oc apply should only be used on resources created declaratively by either oc create --save-config or oc apply. The missing annotation will be patched automatically.
subscription.operators.coreos.com/codeready-workspaces configured
[INFO] Waiting 30s for the new codeready-workspaces operator creation.
error: timed out waiting for the condition on subscriptions/codeready-workspaces
```
Screenshot from 2022-08-24 16-34-16


At the same time subscriptions/codeready-workspaces looked healthy.

- [rollback_after_1-prepare.log](https://github.com/eclipse-che/che-docs/files/9416473/rollback_after_1-prepare.log)
- [codeready-operator-57f9b4465b-8bljc-codeready-operator.log](https://github.com/eclipse-che/che-docs/files/9416474/codeready-operator-57f9b4465b-8bljc-codeready-operator.log)
- [postgres-647fcc5c9d-djw5f-postgres.log](https://github.com/eclipse-che/che-docs/files/9416475/postgres-647fcc5c9d-djw5f-postgres.log)

WDYT about adding notes that rollback after `1-prepare.sh` script is not supported into https://github.com/eclipse-che/che-docs/blob/bd0efcb1b90c270221dbd3f3ef19ebeb95c812dd/modules/administration-guide/examples/proc_che-rolling-the-upgrade-back-from-che-7-49-to-che-7-41-on-openshift.adoc ?

I could also open a ticket for implementing rollback steps after `1-prepare.sh` script.

**Comment by tolusha** [comment]
@dmytro-ndp 
Thank you for checking.
Since the spotted issue is not related to the PR, I suggest to file a dedicated ticket. WDYT?

**Comment by dmytro-ndp** [comment]
@tolusha : you are welcome!
Dedicated ticket: https://issues.redhat.com/browse/CRW-3289

I have also almost successfully rolled back from failed 3-subscribe.sh to initial CRW 2.15.4 instance having 102 OCP users linked with CRW instance. 
rollback.sh script stuck on waiting for codeready Pod to be created: 
[rollback_after_failed_3-subscribe.log](https://github.com/eclipse-che/che-docs/files/9420105/rollback_after_failed_3-subscribe.log)

I had to deploy codeready pod manually by scaling up deployment, and then successfully finished rollback with restored 101th and 102th users and their workspaces.

**Review by dmytro-ndp — APPROVED** [review]
@tolusha : migration CRW 2.15.4-0.1658786714.p > DS 3.0.1 have been tested in airgap OCP 4.10 with 102 CRW users linked with OCP users.

Test passed  :heavy_check_mark: 

Script logs: 
- [1-prepare-3.log](https://github.com/eclipse-che/che-docs/files/9423200/1-prepare-3.log)
- [2-migrate-3.log](https://github.com/eclipse-che/che-docs/files/9423199/2-migrate-3.log)
- [3-subscribe-3.log](https://github.com/eclipse-che/che-docs/files/9423198/3-subscribe-3.log)
- [4-wait-3.log](https://github.com/eclipse-che/che-docs/files/9423196/4-wait-3.log)

File: modules/administration-guide/attachments/migration/1-prepare.sh [hunk 1]
```diff
@@ -61,41 +61,57 @@ getUsers() {
...
-        OPENSHIFT_USER_ID=$(echo "${IDENTITY_PROVIDER}" | jq ".userId" | tr -d "\"")
-        echo "[INFO] Found ${PRODUCT_ID} user: ${USER_ID} and corresponding OpenShift user: ${OPENSHIFT_USER_ID}"
-
-        # Save profile by encoding data and trimming \r\n
-        echo "${USER_ID} ${OPENSHIFT_USER_ID} username:$(echo "${USER_NAME}" | tr -d "\r\n" | base64) email:$(echo "${USER_EMAIL}" | tr -d "\r\n" | base64) firstName:$(echo "${USER_FIRST_NAME}" | tr -d "\r\n" | base64) lastName:$(echo "${USER_LAST_NAME}" | tr -d "\r\n" | base64) " >> "${ALL_USERS_DUMP}"
-      fi
+
+  TOTAL=0
+  FIRST=0
+  MAX=100
```
Comment (ibuziuk) [inline modules/administration-guide/attachments/migration/1-prepare.sh:67]: not 1000 ?

**Review by ibuziuk — APPROVED** [review]
:+1:

**Pull request merged**